packages/create-cloudflare/src/workers.ts

```diff
@@ -1,194 +1,65 @@
-		);
-		await runCommand(
-			[
-				...dlx,
-				"wrangler@3",
-				"init",
-				"--from-dash",
-				ctx.args.existingScript,
-				"-y",
-				"--no-delegate-c3",
-			],
-			{
-				silent: true,
-				cwd: tempdir, // use a tempdir because we don't want all the files
-				env: { CLOUDFLARE_ACCOUNT_ID: ctx.account?.id },
-				startText: "Downloading existing worker files",
-				doneText: `${brandColor("downloaded")} ${dim(
-					`existing "${ctx.args.existingScript}" worker files`
-				)}`,
-			}
-		);
+/**
+ * Update the `wrangler.toml` file for this project by setting the name
+ * to the selected project name and adding the latest compatibility date.
+ */
+export const updateWranglerToml = async (ctx: C3Context) => {
+	if (ctx.template.platform !== "workers") {
+		return;
+	}
 
-		// remove any src/* files from the template
-		for (const filename of await readdir(join(ctx.project.path, "src"))) {
-			await rm(join(ctx.project.path, "src", filename));
-		}
+	const wranglerToml = await readWranglerToml(ctx);
+	const newToml = new MagicString(wranglerToml);
 
-		// copy src/* files from the downloaded worker
-		await cp(
-			join(tempdir, ctx.args.existingScript, "src"),
-			join(ctx.project.path, "src"),
-			{ recursive: true }
+	const compatDateRe = /^compatibility_date\s*=.*/m;
+	if (wranglerToml.match(compatDateRe)) {
+		newToml.replace(
+			compatDateRe,
+			`compatibility_date = "${await getWorkerdCompatibilityDate()}"`
 		);
-
-		// copy wrangler.toml from the downloaded worker
-		await cp(
-			join(tempdir, ctx.args.existingScript, "wrangler.toml"),
-			join(ctx.project.path, "wrangler.toml")
+	} else {
+		newToml.prepend(
+			`compatibility_date = "${await getWorkerdCompatibilityDate()}"`
 		);
 	}
-}
 
-async function updateFiles(ctx: Context) {
-	// Update package.json with project name
-	const pkgJsonPath = resolve(ctx.project.path, "package.json");
-	const pkgJson = readJSON(pkgJsonPath);
-	if (pkgJson.name === "<TBD>") {
-		pkgJson.name = ctx.project.name;
+	const nameRe = /^name\s*=.*/m;
+	if (wranglerToml.match(nameRe)) {
+		newToml.replace(nameRe, `name = "${ctx.project.name}"`);
+	} else {
+		newToml.prepend(`name = "${ctx.project.name}"`);
 	}
-	writeJSON(pkgJsonPath, pkgJson);
 
-	// Update wrangler.toml with name and compat date
-	const wranglerTomlPath = resolve(ctx.project.path, "wrangler.toml");
-	let wranglerToml = readFile(wranglerTomlPath);
-	wranglerToml = wranglerToml
-		.replace(/^name\s*=\s*"<TBD>"/m, `name = "${ctx.project.name}"`)
-		.replace(
-			/^compatibility_date\s*=\s*"<TBD>"/m,
-			`compatibility_date = "${await getWorkerdCompatibilityDate()}"`
-		);
-	writeFile(wranglerTomlPath, wranglerToml);
-}
+	await writeWranglerToml(ctx, newToml.toString());
+};
 
-async function installWorkersTypes(ctx: Context) {
-	if (!ctx.args.ts) {
+/**
+ * Installs the latest version of the `@cloudflare/workers-types` package
+ * and updates the .tsconfig file to use the latest entrypoint version.
```

Comment:
I think it's important to note and keep in mind that this function is imperfect, currently it only updates the `@cloudflare/workers-types` entry (so it assumes that it's already present)

In main it also adds the type if not present, but the implementation there doesn't work with tsconfigs with comments: 
https://github.com/cloudflare/workers-sdk/blob/42ad768693b6b86b7a5835a6394246c01d688c2a/packages/create-cloudflare/src/helpers/command.ts#L484-L497

I'm not saying that we should make this function ultra robust (as that's out of scope here), but I would argue that, if we add such comment, we clarify what updating the tsconfig means and the limitations we have